9.0 mig account journal lock date

robinkeunen opened the pull request:
Allow locking each account journals independently. 

This is a backport of version 10.0 module [account_journal_lock_date](https://github.com/OCA/account-financial-tools/tree/10.0/account_journal_lock_date) by @sbidoul  .

1st pull request for the OCA, best practice or style comments welcome :-)

(2nd try, cf #684 )

sbidoul committed:
account_journal_lock_date

sbidoul committed:
account_journal_lock_date test on travis

The test failed on travis (not on a local install) because the
admin users if part of the Adviser group. Make sure we have the right
group and add a test for that feature too.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
Update account_journal_lock_date.pot

robinkeunen committed:
[ADD] backport of account_journal_lock_date to 9.0

[ADD] README.rst

[REM] data tag from view declaration

pedrobaeza commented:
:clap:

sbidoul commented on account_journal_lock_date/__openerp__.py:
```diff
@@ -0,0 +1,30 @@
+        This is a backport of version 10.0 module of the same name by Stéphane Bidoul.
+
+        https://github.com/OCA/account-financial-tools/tree/10.0/account_journal_lock_date
+    """,
```
No need for a description when there is a README.rst

sbidoul commented on account_journal_lock_date/README.rst:
```diff
@@ -2,6 +2,8 @@
    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
    :alt: License: AGPL-3
 
+.. _account_journal_lock_date: https://github.com/OCA/account-financial-tools/tree/10.0/account_journal_lock_date
```
Instead of addind this, I recommend using the [readme fragments](https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst), which will autogenerate the readme including a github link.

sbidoul commented on account_journal_lock_date/README.rst:
```diff
@@ -19,6 +21,8 @@ Note: the journal lock date is ignored for users that are part of
 `_can_bypass_journal_lock_date` of `account.journal`.
 
+This is a backport of version 10.0 module account_journal_lock_date_ by @sbidoul
+
```
This mention is not necessary

sbidoul commented on account_journal_lock_date/models/account_journal.py:
```diff
@@ -1,19 +1,17 @@
 # -*- coding: utf-8 -*-
-# Copyright 2017 ACSONE SA/NV
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
why removing copyright headed?

robinkeunen replied:
Bad manipulation, it's already fixed in the last fixup. Sorry about that.

sbidoul commented on account_journal_lock_date/models/__init__.py:
```diff
@@ -1,2 +1,4 @@
+# -*- coding: utf-8 -*-
+
```
is the coding comment needed in `__init__.py`?

robinkeunen replied:
I don't think so, my IDE adds it. I'll remove it.

robinkeunen replied:
No, my IDE adds it. I removed it.

sbidoul commented on account_journal_lock_date/models/account_move.py:
```diff
@@ -1,26 +1,29 @@
-                    _("You cannot add/modify entries prior to and "
-                      "inclusive of the journal lock date %s") %
-                    (lock_date, ))
+                raise JournalLockDateError("You cannot add/modify entries "
```
why removing the `_` translation?

robinkeunen replied:
No reason, it's just that I don't know how it works . I'll put it back.

sbidoul commented on account_journal_lock_date/models/account_journal.py:
```diff
@@ -1,19 +1,17 @@
 
     journal_lock_date = fields.Date(
-        string="Lock date",
+        name='Lock Date',
```
why?

robinkeunen replied:
`name` instead of `string` is a mistake, it's fixed. I capitalized _Date_ for consistency with the rest of the code base. But now that I think of it, i might mess up translation. I'll leave it as it is.

sbidoul reviewed (commented):
Thanks for this work. A few comments.

Since there are "cosmetic" changes (which look good), it would be great if you could forward port them to 10 after this one is merged, so as to facilitate maintenance across versions.

robinkeunen committed:
[ADD] readme folder

robinkeunen commented:
Thank you for your comments. I addressed all of them.

I split the readme.rst in _fragments_ in the readme folder. Was it what you expected? If not, can you point to an example?

I'll port the cosmetic changes on version 10 when this PR is done.

robinkeunen commented:
@sbidoul Should we have a look at this at the OCA sprint?

robinkeunen commented:
Hi @sbidoul , I'm doing some cleanup in my PRs. Is this ok to merge?
